code-names-common/src/main/java/com/codenames/enums/GameTurn.java

```diff
@@ -0,0 +1,23 @@
+package com.codenames.enums;
+
+public enum GameTurn {
+    BLUE_MASTER, BLUE_PLAYERS, YELLOW_MASTER, YELLOW_PLAYERS;
+
+    public GameTurn nextTurn(){
+        return switch (this){
+            case BLUE_MASTER -> BLUE_PLAYERS;
+            case BLUE_PLAYERS -> YELLOW_MASTER;
+            case YELLOW_MASTER -> YELLOW_PLAYERS;
+            case YELLOW_PLAYERS -> BLUE_MASTER;
+        };
+    }
+
+    public boolean checkUserTurn(PlayerRole playerRole){
+        GameTurn gameTurn = playerRole.toGameTurn();
+
+        if (gameTurn == null){
```

Comment:
I changed to a simpler code without using Objects.requireNonNullElse because in fact you can do a simple comparison there because if playerRole.toGameTurn() is null, the result will be false